pkg/email/lore/read.go

```diff
@@ -36,19 +36,29 @@ func ReadArchive(repo vcs.Repo, afterCommit string, afterTime time.Time) ([]Emai
 	return ret, nil
 }
 
-func (er *EmailReader) Parse(emails, domains []string) (*email.Email, error) {
+type Email struct {
+	*email.Email
+	HasPatch bool
+}
+
+func (er *EmailReader) Parse(emails, domains []string) (*Email, error) {
 	body, err := er.Read()
 	if err != nil {
 		return nil, err
 	}
+	return emailFromRaw(body, emails, domains)
+}
+
+func emailFromRaw(body []byte, emails, domains []string) (*Email, error) {
 	msg, err := email.Parse(bytes.NewReader(body), emails, nil, domains)
 	if err != nil {
 		return nil, err
 	}
 	// Keep memory consumption low.
-	msg.Body = ""
+	ret := &Email{Email: msg}
```

Comment:
I think it was cheaper to `return &Email{Email: msg}, nil`.
If you want to keep this code, please:
1. Move `// Keep memory consumption low.` closed to the actual `=""` operations.
2. Use `ret.Patch` instead of `mgs.Patch`.